File: Data Structures & Algorithms/binary-tree-diameter/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    private int d = 0;
    public int DiameterOfBinaryTree(TreeNode root) {
        db(root);
        return d;
    }

    public int db(TreeNode root) {
        if (root == null) return 0;
        var left = db(root.left);
        var right = db(root.right);
        d = Math.Max(left + right, d);
        return left > right ? left + 1 : right + 1;
    }
}

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        profit = 0
        for right in range(len(prices)):
            if prices[right] < prices[left]:
                left = right
            else:
                if right == len(prices) - 1 or prices[right + 1] < prices[right]:
                    profit += prices[right] - prices[left]
                    left = right
        return profit